src/sdk/endpoint_stats_collector.js

```diff
@@ -239,6 +239,15 @@ class EndpointStatsCollector {
         });
     }
 
+    async update_semaphore_reports(repotrs){
+        var type = 'ns'
```

Comment:
what is type ns use for? 
yes, it is a lable but for what reasone we define `ns`? 
and even if we should define the lable, can we use a better name? 
and avoid var, use let.